### PR TITLE
fix(ci): align CLI and hardware labels

### LINKED ISSUE #9680: [Feature]: ci: audit remaining CLI and hardware path-label ownership
### Summary

Audit the remaining CLI and hardware source owners that are not covered by the synchronized path-label rules added in #9537.

### Problem statement

#9537 established deterministic `cli` and `hardware` path-label ownership, but its approved review identified additional tracked implementation paths outside those matchers. Examples include:

- `src/memory/cli.rs` for the `zeroclaw memory` command.
- `src/peripherals/mod.rs` for peripheral integration.
- `crates/zeroclaw-api/src/peripherals_traits.rs` for hardware-facing traits.
- `crates/zeroclaw-runtime/src/firmware/` for firmware behavior.

A pull request limited to one of these paths may therefore miss the corresponding retained label even though it changes that surface.

### Proposed solution

1. Inventory the repository paths that own CLI and hardware behavior.
2. Decide which paths belong in the synchronized `cli` or `hardware` contracts and which require narrower documentation.
3. Update `.github/labeler.yml` and `docs/book/src/maintainers/labels.md` together.
4. Add or retain a repeatable matcher matrix covering accepted positive paths and representative negative boundaries with the matching engine pinned by the labeler action.

### Non-goals / out of scope

- Do not rename, retire, or migrate live labels.
- Do not broaden product labels to unrelated implementation paths.
- Do not change runtime, CLI, peripheral, or firmware behavior.

### Acceptance criteria

- The CLI and hardware ownership inventory is explicit and reviewed.
- Every accepted owner path receives the intended synchronized label.
- Representative adjacent paths remain unlabeled when they are outside the contract.
- The maintainer documentation matches the checked-in labeler configuration.
- The matcher matrix is reproducible with the action's pinned matching semantics.

### Risk and rollback

Risk is limited to routing metadata. Over-broad matchers could add misleading labels; incomplete matchers could continue missing labels. Revert the matcher and documentation changes to restore the prior contract.

### Expected routing

This is a bounded CI and maintainer-documentation follow-up to #9537.

### Breaking change?

No.

### Data hygiene checks

- [x] I removed personal or sensitive data from examples and logs.
- [x] I used neutral, project-focused wording and repository paths.

### PR DESCRIPTION
**Pull request by sunlit-deng**
## Summary

- **Base branch:** `master` (all contributions)
- **What changed and why:**
  - Assign `src/memory/cli.rs` to the `cli` path label.
  - Assign the peripheral trait/module paths and the canonical `firmware/**` tree to the `hardware` path label.
  - Add a pinned `minimatch` boundary matrix that proves tracked positive paths and nearby negatives, and run it in CI.
  - Keep the maintainer label table aligned with the executable ownership rules.
- **Scope boundary:** This changes path-label ownership and its deterministic test only. It does not apply labels to existing PRs, change the labeler action, treat the runtime firmware symlink as a source tree, or broaden unrelated subsystem globs.
- **Blast radius:** PR path labeling gains coverage for the memory CLI and hardware abstractions; the new CI script fails if the configured ownership boundaries or tracked positive inventory drift.
- **Linked issue(s):** Closes #9680
- **Labels:** `enhancement`, `ci`, `docs`, `scripts`, `experienced contributor`, `risk:medium`, `size:S`, `type:ci`

### How you can test (when useful)

- **Reviewer testing requested?** N/A — the executable matcher matrix covers the changed rules and boundary cases.

### How I tested

- **CI checks relied on and why they cover this change:** The required CI step runs the same script that parses `.github/labeler.yml`, verifies the pinned `actions/labeler` ref, confirms positive cases are tracked repository paths, and evaluates positive/negative paths with the matching `minimatch` options.
- **Known CI coverage gap, if any:** The live GitHub labeler action was not invoked against a remote PR locally; the matrix is the deterministic local/CI proof of the configured ownership.
- **Commands run and tail output:**

```text
$ bash -n scripts/ci/path_labeler_matrix.test.sh
status: passed

$ bash scripts/ci/path_labeler_matrix.test.sh
Pinned matcher: minimatch 10.2.5 (actions/labeler@bf12e9b00b37c5c0ca2b87b79b2daf7891dbda13 # v7.0.0)
5/5 configured positives matched.
6/6 boundary negatives stayed unmatched.
status: passed

$ git diff --check
status: passed

$ cargo fmt --all -- --check
status: passed

$ cargo clippy --workspace --exclude zeroclaw-desktop --all-targets --features ci-all -- -D warnings
Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.96s
status: passed

$ cargo test --locked
test result: ok; all test targets completed with no failures.
status: passed
```

- **Beyond CI, what did I manually verify?** Compared every positive matrix path with `git ls-files`, confirmed the runtime firmware entry is a tracked symlink rather than a source directory, and reviewed the related open PR file lists for adjacent CI/labels overlap.
- **If any command was intentionally skipped, why:** None.

## Security & Privacy Impact

- New permissions, capabilities, or file system access scope? No
- New external network calls? Yes. The test harness installs exact `minimatch` and `js-yaml` versions when its temporary test environment is absent; scripts are disabled and the packages are used only for local matching/config parsing.
- Secrets / tokens / credentials handling changed? No
- PII, real identities, or personal data in diff, tests, fixtures, or docs? No
- Prompt injection or untrusted model-visible text introduced/changed? No
- If any Yes, describe the risk and mitigation: The temporary npm install is limited to pinned package versions, runs with lifecycle scripts disabled, and is removed after the test process exits.

## Compatibility

- Backward compatible? Yes
- Config / env / CLI surface changed? No
- Rust/MSRV/toolchain floor changed? No
- If backward compatibility is No or either surface/floor question is Yes: exact upgrade steps for existing users: N/A

## Rollback

- **Fast rollback command/path:** After squash merge, run `git revert <squash-merge-sha>` on `master`; this removes the label rules, documentation, and required matrix step together.
- **Feature flags or config toggles:** None.
- **Observable failure symptoms:** The required `Test CLI and hardware path-label ownership` job fails, or affected PRs receive missing or incorrect `cli`/`hardware` labels.

### CODE REVIEW

File: scripts/ci/path_labeler_matrix.test.sh [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    });
+  });
+}
+
+const positives = [
+  ['cli', 'src/memory/cli.rs'],
+  ['cli', 'src/commands/memory.rs'],
+  ['hardware', 'src/peripherals/mod.rs'],
+  ['hardware', 'crates/zeroclaw-api/src/peripherals_traits.rs'],
+  ['hardware', 'crates/zeroclaw-runtime/src/firmware/transport.rs'],
```
Comment (IftekharUddin) [inline scripts/ci/path_labeler_matrix.test.sh]: 🔴 **Blocking:** this positive is not a repository path. `git ls-files` shows `crates/zeroclaw-runtime/src/firmware` is itself a tracked symlink; there is no tracked `transport.rs` below it, and the real firmware files live under the already-covered `firmware/**` root. Because the new glob also does not match the symlink path itself, this synthetic case can pass while proving no real ownership boundary. Please remove the no-op runtime rule/claim or match and test the exact tracked symlink path or paths.

**Review by IftekharUddin — CHANGES_REQUESTED** [review]
@sunlit-deng, I reviewed 7aa78f64c9b9e9adf30d1134e9f6150fc1deaf87 against `master`, including #9680, the current tracked path inventory, the exact pinned `actions/labeler` source, and the completed required checks. There were no earlier formal reviews, inline threads, or top-level review comments to reconcile.

### 🔴 Blocking — the firmware matrix proves a path the repository does not contain

The new positive case uses `crates/zeroclaw-runtime/src/firmware/transport.rs`, but that is not a tracked source path. The repository tracks `crates/zeroclaw-runtime/src/firmware` itself as a symlink, while the actual firmware files are under `firmware/**` and are already covered by the existing rule. A `crates/zeroclaw-runtime/src/firmware/**` glob does not match the tracked symlink path, so the passing synthetic case does not prove that any real changed file receives `hardware`.

**Why this matters:** #9680 requires the ownership inventory and matcher matrix to be evidence for real repository paths. A fabricated positive creates a second, drifting picture of the tree and lets a no-op ownership rule look covered.

**If ignored:** maintainers can reasonably believe the runtime firmware boundary is labeled even though a PR changing the tracked symlink will not receive `hardware`, while real firmware sources continue to be owned only by the pre-existing root rule.

**Required change:** derive this case from the tracked inventory. Either remove the runtime matcher, test, and documentation claim if `firmware/**` is the canonical owner, or explicitly match the real tracked symlink path or paths and test those exact paths.

### 🟡 Warning — the public metadata understates this required-CI change

This PR changes a required CI job and adds a network-installed test dependency, so `risk:medium` is the truthful class under the current label contract; `risk:low` excludes build and CI effects. The body also still says labels are pending and names the branch-head SHA as a rollback target even though this repository permits squash merges only, so that commit will not exist on `master`. These are maintainer-owned cleanup items rather than the reason for my changes-requested verdict, but the label snapshot and medium-risk rollback block need to be corrected before approval.

### 🟢 What looks good — the real ownership additions stay narrow

The `src/memory/cli.rs`, `src/peripherals/mod.rs`, and peripheral-trait additions use exact paths instead of broad subsystem globs, and the maintainer table mirrors those concrete rules. The harness also uses the same direct `minimatch` and `js-yaml` versions as the pinned labeler release and preserves the action's `dot: true` matching option.

I am requesting changes for the false-positive firmware proof above. Once the matrix is grounded in the real tracked paths, this should be a small re-review.

**Review by IftekharUddin — APPROVED** [review]
Reviewed exact head `754722d6a3fa00739767fccf30fba425e84073a5` across the path-label rules, required-CI wiring, executable matcher matrix, maintainer documentation, linked issue, pinned `actions/labeler` implementation, and the complete prior review history.

The prior blocker is resolved. The no-op runtime-firmware symlink glob and documentation claim are removed, every positive matcher case now names a tracked file, and the harness first proves those positives against `git ls-files`. The matrix uses the same `minimatch 10.2.5` matching form as the pinned `actions/labeler` revision and retains focused boundary negatives. I found no remaining correctness, security, privacy, data-integrity, or compatibility blocker at this head.

Hosted CI currently reports 34 successful checks and 2 pending checks, with no failures; pending CI is non-blocking review metadata. The live `risk:low` label understates a required-CI workflow change that adds network-installed test dependencies; the PR body’s `risk:medium` classification is the truthful one. Treating it as medium, this approval is not sufficient by itself under the requested threshold and a distinct second approval is still required before merge.

**Review by Audacity88 — APPROVED** [review]
@sunlit-deng I reviewed current head 754722d6a3fa00739767fccf30fba425e84073a5 against #9680, the path-label source of truth in `.github/labeler.yml`, the maintainer labels guide, the pinned `actions/labeler` dependency versions, and the required CI result.

### ✅ Resolved — Firmware coverage is now grounded in tracked paths

The earlier firmware blocker is fixed. The no-op `crates/zeroclaw-runtime/src/firmware/**` matcher and doc claim are gone, and the matrix now checks each positive case against `git ls-files` before treating it as proof. That keeps the test from passing on fabricated repository paths.

### 🟢 What looks good — The new matcher matrix protects the intended ownership contract

The added `cli` and `hardware` rules are narrow and mirrored in `docs/book/src/maintainers/labels.md`. The test uses the same `minimatch 10.2.5` and `js-yaml 5.1.0` versions as the pinned labeler action, verifies the pinned workflow reference, covers representative negatives, and passed in the required Repository Structure job with all five positives and six negatives behaving as expected.

The live PR label should be corrected from `risk:low` to `risk:medium` because this changes required CI, and the PR body already uses the truthful `risk:medium` classification. With that maintainer metadata cleanup, I found no remaining blocker.
